src/main/java/com/cloudbees/hudson/plugins/folder/AbstractFolder.java

```diff
@@ -0,0 +1,746 @@
+        } finally {
+            SecurityContextHolder.setContext(orig);
+        }
+        super.performDelete();
+    }
+
+    @Override
+    public synchronized void save() throws IOException {
+        if (BulkChange.contains(this)) {
+            return;
+        }
+        super.save();
+        // TODO should this not just be done in AbstractItem?
+        ItemListener.fireOnUpdated(this);
+    }
+
+    /**
+     * Renames this item container.
+     */
+    @Override
+    public void renameTo(String newName) throws IOException {
+        super.renameTo(newName);
+    }
+
+    @Override
+    public synchronized void doSubmitDescription(StaplerRequest req, StaplerResponse rsp) throws IOException, ServletException {
+        getPrimaryView().doSubmitDescription(req, rsp);
+    }
+
+    public void doConfigSubmit(StaplerRequest req, StaplerResponse rsp) throws IOException, ServletException, Descriptor.FormException {
+        checkPermission(CONFIGURE);
+
+        req.setCharacterEncoding("UTF-8");
+        JSONObject json = req.getSubmittedForm();
+
+        BulkChange bc = new BulkChange(this);
+        try {
+            description = json.getString("description");
+            displayName = Util.fixEmpty(json.optString("displayNameOrNull"));
+            if (json.has("viewsTabBar")) {
+                viewsTabBar = req.bindJSON(ViewsTabBar.class, json.getJSONObject("viewsTabBar"));
+            }
+
+            if (json.has("primaryView")) {
+                primaryView = json.getString("primaryView");
+            }
+
+            properties.rebuild(req, json, getDescriptor().getPropertyDescriptors());
+            for (AbstractFolderProperty p : properties) {
+                p.setOwner(this);
+            }
+
+            healthMetrics.replaceBy(req.bindJSONToList(FolderHealthMetric.class, json.get("healthMetrics")));
+
+            icon = req.bindJSON(FolderIcon.class, req.getSubmittedForm().getJSONObject("icon"));
+            icon.setOwner(this);
+
+            submit(req, rsp);
+
+            save();
+            bc.commit();
+        } finally {
+            bc.abort();
+        }
+
+        // TODO boilerplate
+        String newName = json.getString("name");
+        ProjectNamingStrategy namingStrategy = Jenkins.getActiveInstance().getProjectNamingStrategy();
+        if (newName != null && !newName.equals(name)) {
+            // check this error early to avoid HTTP response splitting.
+            Jenkins.checkGoodName(newName);
+            namingStrategy.checkName(newName);
+            rsp.sendRedirect("rename?newName=" + URLEncoder.encode(newName, "UTF-8"));
+        } else {
+            if (namingStrategy.isForceExistingJobs()) {
+                namingStrategy.checkName(name);
+            }
+            FormApply.success(".").generateResponse(req, rsp, this);
+        }
+    }
+
+    protected void submit(StaplerRequest req, StaplerResponse rsp) throws IOException, ServletException, Descriptor.FormException {}
```

Comment:
No, this is a helper method, intended to be overridden (so yes used, in other plugins), not a web method.